### ISSUE TITLE
Smart contract compilation fails after several successful attempts

### ISSUE THREAD

**Issue by shimkiv** [o1-labs/o1js#880]
#### Notes:

- Issue happens for **any** runtime (server, browser);
- SnarkyJS version in use: **0.9.8**;
- Smart contract in use: [E2eZkApp](https://github.com/o1-labs/e2e-zkapp/blob/develop/e2e-zkapp-sc/src/E2eZkApp.ts);
- Compilation operation in use: [example](https://github.com/o1-labs/e2e-zkapp/blob/47641513b6a5b0e6ca6112f1e266b2d9f0a52ceb/e2e-zkapp-ui/utils/ZkAppUtils.ts#L111);
- Hardware in use: **Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz 32.00GB RAM**;
- `zkApp` operation after this issue is unstable and can cause other issues.

#### Steps:

1. Initialize `SnarkyJS` (load zkApp server or browser version).
2. Issue the smart contract compilation several times in a row within the same global context.
  2.1. `4 sequential operations` execution on specified hardware (if it matters).

#### Expected results:

- No issues happen.

#### Actual results:

- Smart contract compilation fails on 4th attempt and exact issue thrown differs depending on some state the zkApp is in (like if the compilation is called within other application methods, [deploy example](https://github.com/o1-labs/e2e-zkapp/blob/47641513b6a5b0e6ca6112f1e266b2d9f0a52ceb/e2e-zkapp-ui/utils/ZkAppUtils.ts#L61), [account state update example](https://github.com/o1-labs/e2e-zkapp/blob/47641513b6a5b0e6ca6112f1e266b2d9f0a52ceb/e2e-zkapp-ui/utils/ZkAppUtils.ts#L199)) but always is thrown during the Smart contract compilation method invocation.

```shell
(3) [0, Array(3), Error: null pointer passed to rust
    at imports.wbg.__wbindgen_throw (http://localhost:9042/javas…]

---

Deploying zkApp for E2E testing...
panicked at 'capacity overflow', /Users/shymkiv/.rustup/toolchains/nightly-2022-09-12-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:518:5
Stack:
Error
    at module.exports.__wbg_new_abda76e883ba8a5f (/Users/shymkiv/projects/o1/mina/src/lib/snarky_js_bindings/snarkyjs/dist/node/_node_bindings/plonk_wasm.cjs:7539:17)

---

Updating zkApp State from "144" to "20736"...
Something went wrong. [
  0,
  [ 248, MlBytes { t: 0, c: 'Jsoo_runtime.Error.Exn', l: 22 }, 238 ],
  RangeError: offset is out of bounds
      at Uint8Array.set (<anonymous>)

---

Deploying zkApp for E2E testing...
panicked at 'caml_bigint_256_of_decimal_string', src/arkworks/bigint_256.rs:72:17
Stack:
Error
    at module.exports.__wbg_new_abda76e883ba8a5f (/Users/shymkiv/projects/o1/mina/src/lib/snarky_js_bindings/snarkyjs/dist/node/_node_bindings/plonk_wasm.cjs:7539:17)
    at console_error_panic_hook::hook::h70346a39261188fe (wasm://wasm/0100b0ae:wasm-function[2046]:0x2fbf1e)

---

panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "FromBytes::read failed" }', src/arkworks/pasta_fp.rs:59:21
Stack:
Error
    at module.exports.__wbg_new_abda76e883ba8a5f (/Users/shymkiv/projects/o1/mina/src/lib/snarky_js_bindings/snarkyjs/dist/node/_node_bindings/plonk_wasm.cjs:7539:17)
```

![image](https://user-images.githubusercontent.com/4096154/235313120-c11fedd8-c2ec-40f5-8e1b-ae59a204ef64.png)

---

[Internal discussion](https://o1-labs.slack.com/archives/C028Q27R8UC/p1682336420164539) with more logs.

**Comment by MartinMinkov** [comment]
I'll take this on and try to reproduce it with the latest o1js versions.

**Comment by MartinMinkov** [comment]
I've set up a local dev environment with [this repo](https://github.com/MartinMinkov/debug-smart-contract-compilation-errors), and I have not been able to reproduce this issue with o1js build: 0.15.2.

For clarity, I used the same [E2EZkapp in the issue](https://github.com/o1-labs/e2e-zkapp/blob/develop/e2e-zkapp-sc/src/E2eZkApp.ts) and deployed it. Then, in the interact script, I [compile 4 times](https://github.com/MartinMinkov/debug-smart-contract-compilation-errors/blob/8f59117e7d9a845a022c7c11272bee35cdc58ef4/src/interact.ts#L72) and create a state-changing transaction with no problem. 

If someone wants to run this, this is the steps I used:

1. git clone git@github.com:MartinMinkov/debug-smart-contract-compilation-errors.git
2. Run `zk lightnet start`
3. Run `zk config` and point the GraphQL endpoint to the running lightnet endpoint.
4. Run `zk deploy`
5. Run `npm run start`

The scripts executes as expected.

**Comment by MartinMinkov** [comment]
Following up, I tried to run the interact script in parallel in 4 different terminal sessions and it worked (the first transaction goes through and the following errors are expected related to `replace_fee`.

![image](https://github.com/o1-labs/o1js/assets/9512405/a1d0505d-7937-4161-b3bd-ff254e015673)

It might be that my hardware is influencing this, perhaps a machine with lower specs could run into the error this issue brings up?

**Comment by shimkiv** [comment]
I can't replicate this issue using the latest `o1js` (`v0.16.0` as of today).
Not sequential nor parallel SC complication replicates the issue.
But the issue replication might also be prevented by couple related issues:
- https://github.com/MinaProtocol/mina/issues/13244
- [Internal thread #1](https://o1-labs.slack.com/archives/C028Q27R8UC/p1693549632081809?thread_ts=1693464492.389019&cid=C028Q27R8UC)
- [Internal thread #2](https://o1-labs.slack.com/archives/C028Q27R8UC/p1698076579096439)
- [Discord](https://discord.com/channels/484437221055922177/1165997249323618314/1165997249323618314)

Considering this I'm not sure what to do with this issue. I'd suggest move it out of scope until mentioned issues/features will be revisited.

cc @MartinMinkov @garwalsh @nicc

**Comment by mitschabaude** [comment]
If it can't be reproduced, close the issue - we have issues for those other problems @shimkiv mentioned, and we can always reopen this one if it reappears

**Comment by shimkiv** [comment]
Thanks @mitschabaude, I'll close it then for time being. 
Can you please link the GH issue here about the circuit methods (`@method`) limits for one SC class? I can't find it.

**Comment by mitschabaude** [comment]
I found this one - https://github.com/o1-labs/o1js/issues/934

**Comment by shimkiv** [comment]
Probably related but I though these are two different issues (the one you mentioned and only `9` supported).
